### PR TITLE
Pd links remove newspaper subscriptions nonuk

### PR DESCRIPTION
**Pull request by paul-daniel-dempsey**
## What are you doing in this PR?
The "Subscriptions" and "Newspaper" links in the navigation should only be visible to users within the UK.

This should be released behind the ab-newProduct 0% AB test so they're visible to the variant only. This AB test has already been set-up and is in use for showing changes to the checkout, you can force the AB test to be visible as so: https://support.theguardian.com/uk/subscribe#ab-newProduct=variant

image

image

image

image

NOTE : 2 menu items small enough to not require side-bar (additional code change reqd?)
image

[**Trello Card**](https://trello.com/c/d3CVEEHj/652-support-site-only-show-subscriptions-and-newspaper-links-in-the-nav-for-the-uk)

## Is this an AB test?
- [x] Yes
- [ ] No

If this is an AB test, PR reviewers should open and check the Optimize test.
[**Optimize Link**](https://optimize.google.com/optimize/home)

### CODE REVIEW

**Comment by paul-daniel-dempsey** [comment]
Have checked for the InternatlisationId meeting = uk. Hope that's right. 
When resizing  the non-uk screens down to iPad/Iphone size, the links header menu is wide enough to display without invoking the  side-bar (see last screenshot). Will a new link be added here in Mk1? If so, the extra space that takes could re-invoke the side-bar. When the screen width is less than the links header menu width the side-bar appears (NOT driven by set iPhone/iPad widths). This formatting behaviour is currently different between the UK and the other regions.

**Comment by GHaberis** [comment]
@paul-daniel-dempsey looking at the screenshots I think we'll need to get a designer to review these changes to the nav

**Comment by paul-daniel-dempsey** [comment]
> @paul-daniel-dempsey looking at the screenshots I think we'll need to get a designer to review these changes to the nav

Yes makes sense, I will look into how to force the side bar to appear in the meantime. Would this be Jade or Helene? If so, shall I request a meeting?

**Comment by GHaberis** [comment]
> > @paul-daniel-dempsey looking at the screenshots I think we'll need to get a designer to review these changes to the nav
> 
> Yes makes sense, I will look into how to force the side bar to appear in the meantime. Would this be Jade or Helene? If so, shall I request a meeting?

I'd mention it in stand-up, but I think Jade should be able to help here

File: support-frontend/assets/components/headers/links/links.tsx [hunk 1]
```diff
@@ -115,10 +115,14 @@ function Links({
 			<ul className="component-header-links__ul" ref={getRef}>
 				{links
 					.filter(({ text }) => {
+						const internationalisationIDValue =
```
Comment (GHaberis) [inline support-frontend/assets/components/headers/links/links.tsx:118]: Currently we're setting `internationalisationIDValue` for each item in the `links` array as we iterate over it, however it's value will always be the same so we may as well only set it once, we could move the variable definition to the top of the `Links` function along with the other variable definitions such as `urlWithoutParams` and only define it once.

Comment (GHaberis) [reply support-frontend/assets/components/headers/links/links.tsx:118]: We could also just save a variable `const isNotUk = internationalisationIDValue !== 'uk'` and then we wouldn't have to duplicate the conditional check `internationalisationIDValue !== 'uk'` multiple times.

Comment (paul-daniel-dempsey) [reply support-frontend/assets/components/headers/links/links.tsx]: Yes agree, much better than repeating the internationalisationID call per links item and isNotUK more readable especially. Have just put the changes in. Just a point of note, with ab-newProduct. When routing/clicking  between links they always go back to the non-variant (ie flipping between UK Subscription/NewsPaper/GWeekly always goes to the non-variant version link). When testing would the UI guys want this? Wish had asked earlier now!

Comment (GHaberis) [reply support-frontend/assets/components/headers/links/links.tsx]: I think when it comes to sharing links for Design/UX to preview the changes it'd be best to create a branch where we enable the `newProduct` AB test and deploy it to `CODE`.

**Review by GHaberis — APPROVED** [review]
Looks good 👍

**Comment by paul-daniel-dempsey** [comment]
[Support site - Only show Subscriptions and Newspaper links in the Nav for the UK](https://trello.com/c/d3CVEEHj/652-support-site-only-show-subscriptions-and-newspaper-links-in-the-nav-for-the-uk)

**Pull request merged**